kdb/kdb.go

```diff
@@ -0,0 +1,151 @@
+package kdb
+
+// #cgo LDFLAGS: -lelektra
+// #include <elektra/kdb.h>
+import "C"
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// KDB is an interface to the elektra library.
+type KDB interface {
+	Open(key Key) error
+	Close(key Key) error
+
+	Get(keySet KeySet, parentKey Key) error
+	Set(keySet KeySet, parentKey Key) error
+
+	// Ensure(contract KeySet, parentKey Key)
+
+	Version() (string, error)
+}
+
+type kdbC struct {
+	handle *C.struct__KDB
+}
+
+// New returns a new KDB instance.
+func New() KDB {
+	return &kdbC{}
+}
+
+// Open creates a handle to the elektra library,
+// this is mandatory to Get / Set Keys.
+func (e *kdbC) Open(key Key) error {
+	k, err := toCKey(key)
+
+	if err != nil {
+		return err
+	}
+
+	handle, ret := C.kdbOpen(k.key)
+
+	if handle == nil {
+		return fmt.Errorf("unable to open kdb: %v", ret)
```

Comment:
Here you should return the full error message you got from Elektra.

Reply:
How do I get the error messages, is there something similiar to elektra.h `elektraErrorDescription()`?

Reply:
see #2875